src/test/http_rest_api_handler_test.cpp

```diff
@@ -0,0 +1,82 @@
+//*****************************************************************************
+// Copyright 2021 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+#include <gtest/gtest.h>
+
+#include "../config.hpp"
+#include "../http_rest_api_handler.hpp"
+#include "../logging.hpp"
+#include "../modelmanager.hpp"
+#include "test_utils.hpp"
+
+#pragma GCC diagnostic ignored "-Wwrite-strings"
+
+static const char* config_1 = R"(
+{
+    "model_config_list": [
+        {
+            "config": {
+                "name": "dummy",
+                "base_path": "/ovms/src/test/dummy"
+            }
+        }
+    ]
+})";
+
+class ModelControlApi : public TestWithTempDir {};
+
+TEST(ModelControlApi, nonExistingConfigFile) {
+    std::filesystem::remove("/tmp/ovms_config_file.json");
+    auto configFile = createConfigFileWithContent(config_1);
+    char* n_argv[] = {"ovms", "--config_path", "/tmp/nonExistingConfig.json", "--file_system_poll_wait_seconds", "0"};
+    int arg_count = 5;
+    ovms::Config::instance().parse(arg_count, n_argv);
+
+    auto handler = ovms::HttpRestApiHandler(10);
+    std::string response;
+    auto status = handler.processModelControlApiRequest(response);
+
+    EXPECT_EQ(status, ovms::StatusCode::FILE_INVALID);
+}
+
+TEST(ModelControlApi, positive) {
```

Comment:
Do we have test to check if reload actually happened? (check the response content?)